lib/node_modules/@stdlib/lapack/base/dlaswp/test/test.dlaswp.js

```diff
@@ -0,0 +1,195 @@
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+/* eslint-disable max-len */
+
+'use strict';
+
+// MODULES //
+
+var tape = require( 'tape' );
+var Int32Array = require( '@stdlib/array/int32' );
+var Float64Array = require( '@stdlib/array/float64' );
+var EPS = require( '@stdlib/constants/float64/eps' );
+var abs = require( '@stdlib/math/base/special/abs' );
+var dlaswp = require( './../lib/dlaswp.js' );
+
+
+// FIXTURES //
+
+var cao = require( './fixtures/column_major_ao.json' );
+var cat = require( './fixtures/column_major_at.json' );
+var ran = require( './fixtures/row_major_an.json' );
+var rao = require( './fixtures/row_major_ao.json' );
+var raolt = require( './fixtures/row_major_aolt.json' );
+
+
+// FUNCTIONS //
+
+/**
+* Tests for element-wise approximate equality.
+*
+* @private
+* @param {Object} t - test object
+* @param {Collection} actual - actual values
+* @param {Collection} expected - expected values
+* @param {number} rtol - relative tolerance
+*/
+function isApprox( t, actual, expected, rtol ) {
+	var delta;
+	var tol;
+	var i;
+
+	t.strictEqual( actual.length, expected.length, 'returns expected value' );
+	for ( i = 0; i < expected.length; i++ ) {
+		if ( actual[ i ] === expected[ i ] ) {
+			t.strictEqual( actual[ i ], expected[ i ], 'returns expected value' );
+		} else {
+			delta = abs( actual[ i ] - expected[ i ] );
+			tol = rtol * EPS * abs( expected[ i ] );
+			t.ok( delta <= tol, 'within tolerance. actual: '+actual[ i ]+'. expected: '+expected[ i ]+'. delta: '+delta+'. tol: '+tol+'.' );
+		}
+	}
+}
+
+
+// TESTS //
+
+tape( 'main export is a function', function test( t ) {
+	t.ok( true, __filename );
+	t.strictEqual( typeof dlaswp, 'function', 'main export is a function' );
+	t.end();
+});
+
+tape( 'the function has an arity of 8', function test( t ) {
+	t.strictEqual( dlaswp.length, 8, 'returns expected value' );
+	t.end();
+});
+
+tape( 'the function throws an error if provided an invalid first argument', function test( t ) {
+	var values;
+	var i;
+
+	values = [
+		'foo',
+		'bar',
+		'beep',
+		'boop'
+	];
+
+	for ( i = 0; i < values.length; i++ ) {
+		t.throws( badValue( values[ i ] ), TypeError, 'throws an error when provided ' + values[ i ] );
+	}
+	t.end();
+
+	function badValue( value ) {
+		return function badValue() {
+			dlaswp( value, cao.N, cao.A, cao.LDA, cao.k1, cao.k2, cao.IPIV, cao.strideIPIV );
+		};
+	}
+});
+
+tape( 'the function performs a series of row interchanges on the matrix `A`, where `A` is an `LDA` by `N` matrix supplied in packed form (column-major, strideIPIV=1)', function test( t ) {
```

Comment:
```suggestion
tape( 'the function performs a series of row interchanges on the matrix `A`, where `A` is an ( `LDA` x `N` ) matrix supplied in packed form (column-major, strideIPIV=1)', function test( t ) {
```

Reply:
Not sure which one is better, open to suggestion.

Reply:
The former. But also, `A` is not a packed matrix, correct?

Reply:
It is not, I'll modify it.